feature/contributor/src/main/java/io/github/droidkaigi/confsched2020/contributor/ui/viewmodel/ContributorsViewModel.kt

```diff
@@ -1,11 +1,87 @@
 package io.github.droidkaigi.confsched2020.contributor.ui.viewmodel
 
+import androidx.lifecycle.LiveData
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.asLiveData
+import androidx.lifecycle.liveData
+import io.github.droidkaigi.confsched2020.ext.combine
+import io.github.droidkaigi.confsched2020.ext.toAppError
+import io.github.droidkaigi.confsched2020.ext.toLoadingState
+import io.github.droidkaigi.confsched2020.model.AppError
+import io.github.droidkaigi.confsched2020.model.Contributor
+import io.github.droidkaigi.confsched2020.model.LoadState
 import io.github.droidkaigi.confsched2020.model.repository.ContributorRepository
+import java.lang.Exception
 import javax.inject.Inject
 
 class ContributorsViewModel @Inject constructor(
     private val contributorRepository: ContributorRepository
 ) : ViewModel() {
-    // TODO Implement
+
+    /**
+     * [ViewState] has to be wrapped to workaround a compile error about type mismatch.
+     * [combine] doesn't seem to understand child classes.
+     */
+    data class UiModel(val viewState: ViewState) {
+        companion object {
+            val DEFAULT = UiModel(ViewState.Loading)
+        }
+    }
+
+    sealed class ViewState {
```

Comment:
Thanks for the various challenges. 🏆 
Explain why I don't provide UiModel in such a shield class.
This is because the actual application needs to be very flexible, such as loading while displaying the cache. Such as display errors, but also display contents.
For that reason, I'm a little nervous about returning in a shielded class. Please let us know what you have.

Reply:
Also, with the current method using ContributorsViewModel.ViewState, if an error occurs from Loaded, the content will be displayed even though there is no data in UiModel. The state of the UiModel and the display state may not match.

Reply:
Thanks for the feedback. I will follow the existing pattern used on other screens.

Re: 
>Also, with the current method using ContributorsViewModel.ViewState, if an error occurs from Loaded, the content will be displayed even though there is no data in UiModel. The state of the UiModel and the display state may not match.

I think the state of the UiModel and what's displayed matches in my approach. The idea is that view renders whatever state the uiModel LiveData emits. A state represents a possible state the view can render independent from what was emitted previously. For example, emitting both an error and `isLoading=true` is supposed to be impossible.

If an error occurs and the ViewModel already has the contributors list, it will emit the list. (or can make it emit the list together with `AppError`, but other screens don't seem to do that) If it has never been able to fetch the contributors list, it will emit the error.